fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/PlayerEntityMixin.java

```diff
@@ -0,0 +1,159 @@
+        }
+    }
+    @Inject(
+        at = @At(
+            value="FIELD",
+            target="net/minecraft/sound/SoundEvents.ENTITY_PLAYER_ATTACK_SWEEP:Lnet/minecraft/sound/SoundEvent;",
+            opcode = Opcodes.GETSTATIC),
+        method = "attack",
+        cancellable = false)
+    public void onPlaySweepSound(CallbackInfo ci)
+    {
+        PlayerEntity pe = (PlayerEntity)(Object)this;
+        ItemExtensions item = ((ItemExtensions)pe.getMainHandStack().getItem());
+        if(item.fabric_getSweepingHitSound() != null)
+        {
+            this.muteSound = true;
+            item.fabric_getSweepingHitSound().playSound(pe.world, pe);
+        }
+    }
+    @Inject(
+        at = @At(
+            value="FIELD",
+            target="net/minecraft/sound/SoundEvents.ENTITY_PLAYER_ATTACK_CRIT:Lnet/minecraft/sound/SoundEvent;",
+            opcode = Opcodes.GETSTATIC),
+        method = "attack",
+        cancellable = false)
+    public void onPlayCritSound(CallbackInfo ci)
+    {
+        PlayerEntity pe = (PlayerEntity)(Object)this;
+        ItemExtensions item = ((ItemExtensions)pe.getMainHandStack().getItem());
+        if(item.fabric_getCriticalHitSound() != null)
+        {
+            this.muteSound = true;
+            item.fabric_getCriticalHitSound().playSound(pe.world, pe);
+        }
+    }
+    @Inject(
+        at = @At(
+            value="FIELD",
+            target="net/minecraft/sound/SoundEvents.ENTITY_PLAYER_ATTACK_STRONG:Lnet/minecraft/sound/SoundEvent;",
+            opcode = Opcodes.GETSTATIC),
+        method = "attack",
+        cancellable = false)
+    public void onPlayStrongSound(CallbackInfo ci)
+    {
+        PlayerEntity pe = (PlayerEntity)(Object)this;
+        ItemExtensions item = ((ItemExtensions)pe.getMainHandStack().getItem());
+        if(item.fabric_getStrongHitSound() != null)
+        {
+            this.muteSound = true;
+            item.fabric_getStrongHitSound().playSound(pe.world, pe);
+        }
+    }
+    @Inject(
+        at = @At(
+            value="FIELD",
+            target="net/minecraft/sound/SoundEvents.ENTITY_PLAYER_ATTACK_WEAK:Lnet/minecraft/sound/SoundEvent;",
+            opcode = Opcodes.GETSTATIC),
+        method = "attack",
+        cancellable = false)
+    public void onPlayWeakSound(CallbackInfo ci)
+    {
+        PlayerEntity pe = (PlayerEntity)(Object)this;
+        ItemExtensions item = ((ItemExtensions)pe.getMainHandStack().getItem());
+        if(item.fabric_getWeakHitSound() != null)
+        {
+            this.muteSound = true;
+            item.fabric_getWeakHitSound().playSound(pe.world, pe);
+        }
+    }
+    @Inject(
+        at = @At(
+            value="FIELD",
+            target="net/minecraft/sound/SoundEvents.ENTITY_PLAYER_ATTACK_NODAMAGE:Lnet/minecraft/sound/SoundEvent;",
+            opcode = Opcodes.GETSTATIC),
+        method = "attack",
+        cancellable = false)
+    public void onPlayNoDamageSound(CallbackInfo ci)
+    {
```

Comment:
again, this wont pass checkstyle

Reply:
Yeah i forgot about the checkstyle and i had the plugin off. I'm working on them now